central/role/datastore/datastore_impl.go

```diff
@@ -103,7 +103,7 @@ func (ds *dataStoreImpl) UpdateRole(ctx context.Context, role *storage.Role) err
 	if err := rolePkg.ValidateRole(role); err != nil {
 		return errors.Wrap(errox.InvalidArgs, err.Error())
 	}
-	if err := verifyNotDefaultRole(role.GetName()); err != nil {
+	if err := verifyNotDefaultRole(role); err != nil {
```

Comment:
That does not work in case the user decides to modify default role and sends it without `traits`

Reply:
I think to be on the safe side here, I've re-added the conditions for the `IsDefault..` functions to conditionally check the name as well, covering the case the traits field isn't set.

Reply:
Good thinking! That way we keep the existing restriction + it's impossible to create a role with "default" origin